mongodbatlas/resource_mongodbatlas_federated_database_instance.go

```diff
@@ -529,151 +533,126 @@ func newStores(d *schema.ResourceData) []*matlas.DataFederationStore {
+		additionalStorageClasses[i] = additionalStorageClassFromConfig.(string)
 	}
 
 	return additionalStorageClasses
 }
 
-func newReadPreference(storeFromConfMap map[string]interface{}) *matlas.ReadPreference {
+func newReadPreference(storeFromConfMap map[string]interface{}) *admin.DataLakeAtlasStoreReadPreference {
 	readPreferenceFromConf, ok := storeFromConfMap["read_preference"].([]interface{})
 	if !ok || len(readPreferenceFromConf) == 0 {
 		return nil
 	}
 	readPreferenceFromConfMap := readPreferenceFromConf[0].(map[string]interface{})
-	return &matlas.ReadPreference{
-		Mode:                readPreferenceFromConfMap["mode"].(string),
-		MaxStalenessSeconds: int32(readPreferenceFromConfMap["max_staleness_seconds"].(int)),
-		TagSets:             newTagSets(readPreferenceFromConfMap),
+	return &admin.DataLakeAtlasStoreReadPreference{
+		Mode:                stringPtr(readPreferenceFromConfMap["mode"].(string)),
+		MaxStalenessSeconds: intPtr(readPreferenceFromConfMap["max_staleness_seconds"].(int)),
 	}
 }
 
-func newTagSets(readPreferenceFromConfMap map[string]interface{}) []*matlas.TagSet {
-	tagSetsFromConf, ok := readPreferenceFromConfMap["tag_sets"].([]interface{})
-	if !ok || len(tagSetsFromConf) == 0 {
-		return nil
-	}
-
-	tagSets := make([]*matlas.TagSet, len(tagSetsFromConf))
-	for i, tagSetFromConf := range tagSetsFromConf {
-		storeFromConfMap := tagSetFromConf.(map[string]interface{})
-		tagSets[i] = &matlas.TagSet{
-			Name:  storeFromConfMap["name"].(string),
-			Value: storeFromConfMap["value"].(string),
-		}
-	}
-
-	return tagSets
-}
-
-func newDataFederationDatabase(d *schema.ResourceData) []*matlas.DataFederationDatabase {
+func newDataFederationDatabase(d *schema.ResourceData) []admin.DataLakeDatabaseInstance {
 	storageDBsFromConf := d.Get("storage_databases").(*schema.Set).List()
 	if len(storageDBsFromConf) == 0 {
 		return nil
 	}
 
-	dbs := make([]*matlas.DataFederationDatabase, len(storageDBsFromConf))
+	dbs := make([]admin.DataLakeDatabaseInstance, len(storageDBsFromConf))
 	for i, storageDBFromConf := range storageDBsFromConf {
 		storageDBFromConfMap := storageDBFromConf.(map[string]interface{})
-		dbs[i] = &matlas.DataFederationDatabase{
-			Name:                   storageDBFromConfMap["name"].(string),
-			MaxWildcardCollections: int32(storageDBFromConfMap["max_wildcard_collections"].(int)),
+
+		dbs[i] = admin.DataLakeDatabaseInstance{
+			Name:                   stringPtr(storageDBFromConfMap["name"].(string)),
+			MaxWildcardCollections: intPtr(storageDBFromConfMap["max_wildcard_collections"].(int)),
 			Collections:            newDataFederationCollections(storageDBFromConfMap),
 		}
 	}
 
 	return dbs
 }
 
-func newDataFederationCollections(storageDBFromConfMap map[string]interface{}) []*matlas.DataFederationCollection {
+func newDataFederationCollections(storageDBFromConfMap map[string]interface{}) []admin.DataLakeDatabaseCollection {
```

Comment:
```suggestion
func newDataFederationCollections(storageDBFromConfMap map[string]interface{}) []*admin.DataLakeDatabaseCollection {
```

Reply:
`[]admin.DataLakeDatabaseCollection` is the required data type by the SDK